holoviews/plotting/bokeh/callbacks.py

```diff
@@ -330,6 +350,75 @@ def _get_stream_handle_ids(self, handles):
         return stream_handle_ids
 
 
+    def on_change(self, attr, old, new):
+        """
+        Process change events adding timeout to process multiple concerted
+        value change at once rather than firing off multiple plot updates.
+        """
+        self._event_queue.append((attr, old, new))
+        if self.trigger not in self.plot.document._session_callbacks:
+            self.plot.document.add_timeout_callback(self.trigger, 50)
+
+
+    def on_event(self, event):
+        """
+        Process bokeh UIEvents adding timeout to process multiple concerted
+        value change at once rather than firing off multiple plot updates.
+        """
+        self._event_queue.append((event))
+        if self.trigger not in self.plot.document._session_callbacks:
+            self.plot.document.add_timeout_callback(self.trigger, 50)
```

Comment:
The only difference between these is how the queue is used. I think it is possible to simplify. e.g

``def debounce(change_spec=None, event_spec=None):
   if change_spec: # Use trigger_change and add to queue appropriately
        callback = self.trigger_change
   elif event_spec: # Use trigger_change and add to queue appropriately
       callback = self.trigger_event
   self.plot.document.add_timeout_callback(callback, 50)
```